### ISSUE TITLE
[BUG][JAVA] Multi-module spring-boot app gives errors about ApplicationContext and "multiple SLF4J bindings" warning  

### ISSUE THREAD

**Issue by LeonardGC** [OpenAPITools/openapi-generator#12375]
### Description:

**TL;DR.**
I made a simple multi-module maven spring-boot app but building this fails with an error: Failed to load ApplicationContext
caused by: LoggerFactory is not a Logback LoggerContext but Logback is on the classpath. ....

**Details:**
I have made a spring boot app with 2 modules that build correclty.
Adding the dependency of org.openapitools:openapi-generator:5.4.0 results in an error: Failed to load ApplicationContext (for the test unit) and was apparently caused by LoggerFactory is not a Logback LoggerContext but Logback is on the classpath. ....

A few lines above this error I have this warning:
  SLF4J: Class path contains multiple SLF4J bindings.

Note that the addition of the dependency  was enough for the exception to occur. No generation has been done!

**_I think the addition of this dependency should not result in an error in the app, especially because it seems related to the logging framework._** 

### Facts
 org.openapitools:openapi-generator:5.4.0
 spring-boot application: 2.6.7 (from initializr converted to multi-module) 

### Code

5 Project files and a directory tree

- [Tree.txt](https://gist.github.com/LeonardGC/0fb9177e5a56cc701b2e7069ffc852da#file-tree-txt): shows the directory structure
- [\pom.xml](https://gist.github.com/LeonardGC/0fb9177e5a56cc701b2e7069ffc852da#file-pom-xml)
- [\OpenApi\pom.xml](https://gist.github.com/LeonardGC/0fb9177e5a56cc701b2e7069ffc852da#file-openapi-pom-xml)
- [\server\pom.xml](https://gist.github.com/LeonardGC/0fb9177e5a56cc701b2e7069ffc852da#file-server-pom-xml)
- [\server\src\main\java\com\example\demo\DemoApplication.java](https://gist.github.com/LeonardGC/0fb9177e5a56cc701b2e7069ffc852da#file-server-src-main-java-com-example-demo-demoapplication-java)
- [\server\src\test\java\com\example\demo\DemoApplicationTests.java](https://gist.github.com/LeonardGC/0fb9177e5a56cc701b2e7069ffc852da#file-server-src-test-java-com-example-demo-demoapplicationtests-java)

**Workaround**
I was able to workaround the issue by excluding the slf4j-simple artifact from the generator (see also commented code in [OpenApi\pom.xml](https://gist.github.com/LeonardGC/0fb9177e5a56cc701b2e7069ffc852da#file-openapi-pom-xml)):

**Comment by harel-biton** [comment]
@wing328 I think this one can also be closed due to https://github.com/OpenAPITools/openapi-generator/pull/18881
